pkg/segment/reader/segread/multicolreader.go

```diff
@@ -89,6 +90,7 @@ func initNewMultiColumnReader(segKey string, colFDs map[string]*os.File, blockMe
 		allColsReverseIndex: colRevserseIndex,
 		timeStampKey:        tsKey,
 		segKey:              segKey,
+		maxColIdx:           -1,
```

Comment:
nit: it looks like `-1` is a special value to indicate it's invalid. This would be a great place to use `utils.Option[int]` :) But I don't think it's necessary to change the code since you already have it this way